packages/main/src/components/VariantManagement/index.tsx

```diff
@@ -59,101 +64,107 @@ const styles = ({ parameters }: JSSTheme) => ({
-
-  state = {
-    selectedKey: this.initialSelectedKey,
-    open: false
-  };
-
-  _getItemByKey(key, items) {
-    return items.filter((item) => item.key === key)[0];
-  }
-
-  private handleCancelButtonClick = () => {
-    this.setState({ open: false });
-  };
-
-  private handleAfterOpen = () => {
-    this.setState({ open: true });
-  };
-
-  private handleVariantItemSelect = (event) => {
-    const selectedKey = event.getParameter('item').getAttribute('data-key');
-    const { variantItems } = this.props as VariantManagementInternalProps;
-    const selectedItem = this._getItemByKey(selectedKey, variantItems) || variantItems[0];
-    this.setState({ selectedKey });
-    this.props.onSelect(Event.of(this, event.getOriginalEvent(), { selectedItem }));
-    if (this.props.closeOnItemSelect) {
-      const openPopover = !this.state.open;
-      this.setState({ open: openPopover });
-    }
-  };
-
-  private get variantManagementButton() {
-    const { classes, variantItems, level } = this.props as VariantManagementInternalProps;
-    const { selectedKey } = this.state;
-    const selectedItem = this._getItemByKey(selectedKey, variantItems) || variantItems[0];
-
-    return (
-      <div className={classes.VariantManagement}>
-        <span className={classes.VariantManagementText}>
-          <Title level={level}>{selectedItem.label}</Title>
-        </span>
-        <Button design={ButtonDesign.Transparent} icon={'navigation-down-arrow'} />
-      </div>
+const useStyles = createUseStyles<JSSTheme, keyof ReturnType<typeof styles>>(styles, { name: 'Variant Management' });
+
+const VariantManagement: FC<VariantManagementPropTypes> = forwardRef(
+  (props: VariantManagementPropTypes, ref: Ref<any>) => {
+    const {
+      variantItems,
+      popupTitle,
+      className,
+      style,
+      tooltip,
+      placement,
+      level,
+      initialSelectedKey,
+      onSelect,
+      closeOnItemSelect,
+      disabled
+    } = props;
+    const classes = useStyles();
+    const [open, setOpen] = useState(false);
+    const [selectedKey, setSelectedKey] = useState(
+      initialSelectedKey ? initialSelectedKey : variantItems.length ? variantItems[0].key : null
     );
-  }
-
-  private getVariantManagementFooterButtons = () => {
-    const { classes } = this.props as VariantManagementInternalProps;
-    return [
-      <Button
-        className={classes.footer}
-        key="btn-cancel"
-        onClick={this.handleCancelButtonClick}
-        design={ButtonDesign.Emphasized}
-      >
-        Cancel
-      </Button>
-    ] as Array<ReactElement<ButtonPropTypes>>;
-  };
-
-  render() {
-    const { variantItems, popupTitle, className, style, tooltip, innerRef } = this
-      .props as VariantManagementInternalProps;
-    const { selectedKey } = this.state;
 
     if (!variantItems || variantItems.length < 1) {
       return null;
     }
 
+    useEffect(() => {
+      setSelectedKey(initialSelectedKey);
```

Comment:
you need to check whether initialSelectedKey is truthy, or you might set the selectedKey to null/undefined after you´ve initialized it to variantItems[0].key

Reply:
fixed